src/disjoint_mut.rs

```diff
@@ -80,6 +100,21 @@ pub struct DisjointImmutGuard<'a, T: AsMutPtr, V: ?Sized> {
     range: (Bound<usize>, Bound<usize>),
 }
 
+impl<'a, T: AsMutPtr> DisjointImmutGuard<'a, T, [u8]> {
+    pub fn cast<V: AsBytes + FromBytes>(self) -> DisjointImmutGuard<'a, T, V> {
```

Comment:
Why do we need the `AsBytes` constraint here?  `fn ref_from` is just on `FromBytes`.

Reply:
copied that from the mut version, thanks.